apps/server/src/modules/oauth/controller/api-test/oauth-sso.api.spec.ts

```diff
@@ -315,21 +363,111 @@ describe('OAuth SSO Controller (API)', () => {
 			});
 		});
 
-		describe('when a faulty query is passed', () => {
-			it('should redirect to the login page with an error', async () => {
-				const { user, system } = await setup();
-				currentUser = mapUserToCurrentUser(user);
-				const { state, cookies } = await setupSessionState(system.id);
-				const clientUrl: string = Configuration.get('HOST') as string;
+		describe('when invalid request', () => {
+			it('should throw UnprocessableEntityException', async () => {
+				const { targetSystem, sourceUser, sourceSystem } = await setup();
+				currentUser = mapUserToCurrentUser(sourceUser, undefined, undefined);
+				const { state, cookies } = await setupSessionState(targetSystem.id, true);
 				const query: AuthorizationParams = new AuthorizationParams();
+				query.error = SSOAuthenticationError.INVALID_REQUEST;
+				query.state = state;
+
+				await request(app.getHttpServer()).get(`/sso/oauth/migration`).set('Cookie', cookies).query(query).expect(422);
+			});
+		});
+
+		describe('when invalid request', () => {
+			it('should redirect to the general migration error page', async () => {
+				const { targetSystem, sourceUser, sourceSystem } = await setup();
+				currentUser = mapUserToCurrentUser(sourceUser, undefined, sourceSystem.id);
+				const { state, cookies } = await setupSessionState(targetSystem.id, true);
+				const baseUrl: string = Configuration.get('HOST') as string;
+				const query: AuthorizationParams = new AuthorizationParams();
+				query.error = SSOAuthenticationError.INVALID_REQUEST;
 				query.state = state;
 
 				await request(app.getHttpServer())
 					.get(`/sso/oauth/migration`)
 					.set('Cookie', cookies)
 					.query(query)
 					.expect(302)
-					.expect('Location', `${clientUrl}/login?error=sso_auth_code_step`);
+					.expect(
+						'Location',
+						`${baseUrl}/migration/error?sourceSystem=${sourceSystem.id}&targetSystem=${targetSystem.id}`
+					);
+			});
+		});
+
+		describe('when schoolnumbers mismatch', () => {
+			it('should redirect to the login page with an schoolnumber mismatch error', async () => {
+				const { targetSystem, sourceUser, targetUser, sourceSystem } = await setup();
+				currentUser = mapUserToCurrentUser(sourceUser, undefined, sourceSystem.id);
+				const { state, cookies } = await setupSessionState(targetSystem.id, true);
```

Comment:
If you resort the setup method from top, some of this lines can be moved to the setup of the senario.